centreon/src/Core/Dashboard/Application/UseCase/PartialUpdateDashboard/PartialUpdateDashboardPanelsDifference.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+/*
+ * Copyright 2005 - 2023 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\Dashboard\Application\UseCase\PartialUpdateDashboard;
+
+use Assert\AssertionFailedException;
+use Core\Dashboard\Application\Exception\DashboardException;
+use Core\Dashboard\Application\UseCase\PartialUpdateDashboard\Request\PanelRequestDto;
+use Core\Dashboard\Domain\Model\DashboardPanel;
+use Core\Dashboard\Domain\Model\NewDashboardPanel;
+
+/**
+ * The goal of this class is to know how to make the different repository calls depending on update data.
+ */
+final class PartialUpdateDashboardPanelsDifference
+{
+    /** @var array<NewDashboardPanel> */
+    private readonly array $panelsToCreate;
+
+    /** @var array<DashboardPanel> */
+    private readonly array $panelsToUpdate;
+
+    /** @var array<int> */
+    private readonly array $panelIdsToDelete;
+
+    /**
+     * @param array<int> $panelIdsFromRepository
+     * @param array<PanelRequestDto> $panelsFromRequest
+     *
+     * @throws DashboardException|AssertionFailedException
+     */
+    public function __construct(array $panelIdsFromRepository, array $panelsFromRequest)
+    {
+        [$this->panelsToCreate, $this->panelsToUpdate, $this->panelIdsToDelete]
+            = $this->compute($panelIdsFromRepository, $panelsFromRequest);
```

Comment:
The compute method should not be called directly by constructor

Reply:
So I should create lazy getters ?
Because the computation of one of each 3 getter depend on the others.
If i made 3 computations in each getter, this won't be nice from a performance point of view.

Can you suggest me something ?